src/TuDa.CIMS.Api/Repositories/PurchaseRepository.cs

```diff
@@ -0,0 +1,164 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TuDa.CIMS.Api.Database;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Repositories;
+
+public class PurchaseRepository : IPurchaseRepository
+{
+    private readonly CIMSDbContext _context;
+
+    public PurchaseRepository(CIMSDbContext context)
+    {
+        _context = context;
+    }
+
+    /// <summary>
+    /// Returns all purchases from the database.
+    /// </summary>
+    /// <param name="workingGroupId">the specific ID of the workinggroup</param>
+    /// <returns></returns>
+    public async Task<IEnumerable<Purchase>> GetAllAsync(Guid workingGroupId)
+    {
+        return await _context.WorkingGroups
+            .Where(i => i.Id == workingGroupId)
+            .SelectMany(i => i.Purchases)
+            .Include(i => i.Buyer)
+            .Include(i => i.Entries)
```

Comment:
🔧 🤔 I think we need to use `ThenInclude` to also include the `AssetItem` of the Entry